poseidon2/src/round_numbers.rs

```diff
@@ -1,24 +1,42 @@
+// As the security analysis of Poseidon2 is identical to that of Poseidon,
+// the relevant constraints regarding the number of full/partial rounds required can be found in
+// the original paper: https://eprint.iacr.org/2019/458.pdf and the associated codebase:
+// https://extgit.iaik.tugraz.at/krypto/hadeshash (See generate_params_poseidon.sage)
+
+// These constraints are broken down into 6 equations:
+// statistical, interpolation, groebner 1, 2, 3 and
+// an extra constraint coming from the paper https://eprint.iacr.org/2023/537.pdf.
+
+// For our parameters (M = 128, p > 2^30, WIDTH = t >= 8, D = alpha < 12),
+// the statistical constraint always simplifies to requiring RF >= 6.
+// Additionally p does not appear in Groebner 3 or the constraint coming from https://eprint.iacr.org/2023/537.pdf.
+// The remaining 3 constraints all can be rearranged into the form:
+// F(RF, RP) >= G(p) where G is a function which is non-decreasing with respect to p.
+
+// Thus, if some tuple (M, p, WIDTH, D, RF, RP) satisfies all constraints, then so will
+// the tuple (M, q, WIDTH, D, RF, RP) for any 2^30 < q < p.
+// Moreover if RF, RP are the "optimal" round numbers (Optimal meaning minimising the number of s-box operations we need to perform)
+// for two tuples (M, p, WIDTH, D) and (M, q, WIDTH, D), then
+// they will also be optimal for (M, r, WIDTH, D) for any q < r < p.
+
 // We compute the optimal required number of external (full) and internal (partial) rounds using:
-// See https://github.com/0xPolygonZero/hash-constants
-// Note the the number of rounds required is the same for fields of approximately the same size.
+// https://github.com/0xPolygonZero/hash-constants
```

Comment:
Nit: Better to put the whole URL, as there are a bunch of other files in that repo.
```suggestion
// https://github.com/0xPolygonZero/hash-constants/blob/master/calc_round_numbers.py
```